src/acceleration/impl/ResidualSumPreconditioner.cpp

```diff
@@ -67,19 +73,39 @@ void ResidualSumPreconditioner::_update_(bool                   timeWindowComple
     }
 
     offset = 0;
-    for (size_t k = 0; k < _subVectorSizes.size(); k++) {
-      if (not math::equals(_residualSum[k], 0.0)) {
-        for (size_t i = 0; i < _subVectorSizes[k]; i++) {
-          _weights[i + offset]    = 1 / _residualSum[k];
-          _invWeights[i + offset] = _residualSum[k];
+    // Check if the new scaling weights are more or less than 1 order of magnitude from the previous weights
```

Comment:
The "1" here is now configurable (`updatePreconLimit`), right? Also in the Debug statement below.